Reference/Routing/custom-controllers-v9.md

```diff
@@ -0,0 +1,421 @@
+---
+versionFrom: 9.0.0
+verified-against: beta-2
+meta-title: Umbraco Route Hijacking
+meta.Description:  Use a custom controller to handle and control incoming requests for content pages based on a specific Document Type
+---
+
+# Custom MVC controllers (Umbraco Route Hijacking)
+
+_Use a custom controller to handle and control incoming requests for content pages based on a specific Document Type_
+
+## What is Umbraco Route Hijacking?
+
+By default, all front end requests to an Umbraco site are auto-routed via the 'Index' action of a core Controller: `Umbraco.Cms.Web.Common.Controllers.RenderController`. This core controller handles the incoming request, builds the associated PublishedContent model, and passes this to the appropriate Umbraco Template/MVC View.
+
+It is however possible to implement a custom Controller to replace this default implementation to give complete control over this execution.
+
+For example:
+
+- To enrich the view model passed to the template with additional properties (from other published content items or outside Umbraco)
+- To implement serverside paging
+- To implement any custom/granular security
+- To return alternative templates depending on some custom business logic
+
+This replacement of the default controller can either be made 'globally' for all requests (see last example) or by *'hijacking'* requests for types of pages based on their specific Document Type following a this controller naming convention: `[DocumentTypeAlias]Controller`.
+
+## Creating a custom controller
+
+### Example: Hijacking route requests to a 'product' page
+
+In the following example, imagine an Umbraco site with a set of 'product' pages created from a Document Type called 'Product Page' with an alias 'productPage'.
+
+Create a custom locally declared controller in the Umbraco web application project (or App_Code folder for website projects) named 'ProductPageController'.
```

Comment:
We might want to remove the reference to App_Code here since App_Code is no longer a thing, so it might be confusing

Reply:
@nikolajlauridsen Good point! Thanks!